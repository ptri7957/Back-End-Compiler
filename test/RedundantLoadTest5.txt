( (branches (a)
    (0 (lc r1 3)
       (st a r1)
       (lc r2 5)
       (st b r2)
       (lc r3 4)
       (st d r3)
       (lc r4 100)
       (st x r4)
       (ld r5 a)
       (ld r6 b)
       (gt r7 r5 r6)
       (st a r7)
       (ld r8 a)
       (br r8 1 2) )
    (1 (ld r9 a)
       (ld r10 b)
       (add r11 r9 r10)
       (st c r11)
       (lc r12 2)
       (st d r12)
       (ld r13 d)
       (ret r13) )
    (2 (lc r14 4)
       (st c r14)
       (ld r15 b)
       (ld r16 d)
       (mul r17 r15 r16)
       (ld r18 c)
       (add r19 r17 r18)
       (st b r19)
       (ld r20 b)
       (ret r20) ) ) )