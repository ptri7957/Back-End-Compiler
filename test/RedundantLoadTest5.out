( (branches (a)
    (0 (lc r1 3)
       (st a r1)
       (lc r2 5)
       (st b r2)
       (lc r3 4)
       (st d r3)
       (lc r4 100)
       (st x r4)
       (ld r5 a)
       (ld r6 b)
       (gt r7 r5 r6)
       (st a r7)
       (ld r8 a)
       (br r8 1 2) )
    (1 (add r11 r7 r2)
       (st c r11)
       (st d r12)
       (ld r13 d)
       (ret r13) )
    (2 (lc r14 4)
       (st c r14)
       (st b r19)
       (ld r20 b)
       (ret r20) ) ) )